bite-size-notes/week-5-e-SCvx-and-GuSTO-SCP.md

```diff
@@ -0,0 +1,127 @@
+
+The optimization problem is solved through the following:
+> 1. **Formulation:** specifications of how the functions to be minimized and constraints to be satisfied are expressed mathematically.
+> 2. **Discretization:** approximation of the infinite-dimensional state and control signals by a finite-dimensional set of basis functions.
+> 3. **Numerical Optimization:** iterative computation of an optimal solution of the discretized problem.
+
+In terms of efficiency and reliability for numerical optimization, discretized convex problems are the way to go, which is the main motivation for **Convex Programming** or **Convex Optimization**. However, you may find that most problems in the real world are **nonlinear (nonconvex)**, which leads to the topic of **Convexification** and **SCP** for trajectory generation of nonlinear dynamic systems based on convex optimization. The main idea of SCP is iterative convex approximation as the name suggests. The general outline of an SCP algorithm is shown in **Fig. 1** where it starts with an initial trajectory guess that goes through an iteration scheme that optimizes the trajectory until it converges into a feasible solution at which the iteration is stopped. With SCP, safe and reliable trajectories can be generated for nonconvex functions.
+
+![SCP Block Diagram](figs/SCP_diagram.png)
+**Figure 1:** Block diagram of a typical SCP algorithm acquired from [1].
+
+An algorithm hierarchy in **Fig. 2** shows where SCP lies. Typically, the convex solver in layer 2 of **Fig. 1** iteratively calls an **Interior Point Method (IPM)** algorithm from layer 3 in **Fig. 2** that solves a convex problem with linear equality and convex inequality constraints as a sequence of linear equality-constrained problems and thus, IPM iteratively calls the algorithm in layer 2 in **Fig. 2**.
+
+![SCP Block Diagram](figs/SCP_hierarchy.png)
+**Figure 2:** Algorithm hierarchy also acquired from [1].
+
+## Preliminaries
+**Formulation**
+The objective of SCP is to solve continuous-time-optimal control problems in the following form:
+|Equation  |#      |
+|----------|-------|
+|$$\min_{u, p} J(x, u, p),$$| (1a)|
+|$$s.t. \quad \dot{x}(t) = f(t, x(t), u(t), p),$$| (1b)|
+|$$(x(t), p) \in \mathcal{X}(t),$$| (1c)|
+|$$(u(t), p) \in \mathcal{U}(t),$$| (1d)|
+|$$s(t, x(t), u(t), p) \leq 0,$$| (1e)|
+|$$g_{ic}(x(0), p) = 0,$$| (1f)|
+|$$g_{tc}(x(1), p) = 0,$$| (1g)|
```

Comment:
The space between equations is a bit too wide, making them somewhat difficult to read. Please consider using the following markdown notation instead:
```
$$
\begin{align}
\min_{u, p}& J(x, u, p) \\ 
s.t. \quad &\dot{x}(t) = f(t, x(t), u(t), p) \\
&(x(t), p) \in \mathcal{X}(t)\\
&(u(t), p) \in \mathcal{U}(t)\\
&s(t, x(t), u(t), p) \leq 0\\
&g_{ic}(x(0), p) = 0\\
&g_{tc}(x(1), p) = 0
\end{align}
$$
```

Reply:
equation format accepted. I added labels using \tag{label}